### PR TITLE
Don`t Merge Yet - Add Caffe2, PyTorch and GPU support(using Accelerators)

### LINKED ISSUE #16: Prepare TensorFlow and Caffe sample GPU jobs.
> Note: Currently GPU workloads on FfDL only works with Feature gate accelerator.

We need to prepare for some sample TensorFlow and Caffe jobs that uses GPU.

* To run TensorFlow and Caffe GPU workloads, we need to build the LCM image with bvlc/caffe:gpu and tensorflow/tensorflow:latest-gpu enable under `lcm/service/lcm/container_helper_extensions.go`

### LINKED ISSUE #17: Adding support for Caffe2 and PyTorch
@Tomcli based on your investigation what`s needed on this side?

### PR DESCRIPTION
**Pull request by Tomcli**
- Add new community CPU images for Caffe2 and PyTorch
- Add sample CPU jobs for Caffe2 and PyTorch
- Add other community learner images (including GPU images)
- Add sample GPU jobs for TensorFlow and Caffe

- [ ] We need to rebuild ffdl-lcm in order to reflect the new changes for the FfDL DockerHub images.

For GPU usage, temporary solution will be available at [gpu-guide](https://github.com/IBM/FfDL/blob/new-frameworks/docs/gpu-guide.md).

This PR fixes #16 , #17 , and #20 .

### CODE REVIEW

**Comment by Tomcli** [comment]
The new commit allow LCM to use different learner images (including GPU images). Now users can use the Framework version section to select the [Framework version](https://github.com/IBM/FfDL/blob/new-frameworks/docs/user-guide.md#1-supported-deep-learning-frameworks) of their choice. 

Please note that the example jobs are configured to pass for every framework on the list. Thus, it might not demonstrate the real world workload. (e.g. For Caffe2 the example is not performing epochs since the latest CPU image is not capable to do that yet.)

To run workloads using GPU, users must satisfy the [following prerequisites](https://github.com/IBM/FfDL/blob/new-frameworks/docs/gpu-guide.md). Currently, using feature gate `Accelerators` is the only option we have. 

- [ ] **Please do not merge this PR until we upload the new version of ffdl-lcm image on DockerHub since the new example jobs is not backward compatible with the old ffdl-lcm image.**

File: etc/examples/caffe-gpu-model/lenet_solver.prototxt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# The train/test net protocol buffer definition
```
Comment (animeshsingh) [inline etc/examples/caffe-gpu-model/lenet_solver.prototxt]: Do we need to duplicate this whole file for a minimal change? Can we ask users to change 1/2 params manually - or give a script?
https://github.com/IBM/FfDL/blob/master/etc/examples/caffe-model/lenet_solver.prototxt

File: etc/examples/caffe-gpu-model/lenet_train_test.prototxt [hunk 2]
```diff
@@ -0,0 +1,168 @@
+name: "LeNet"
```
Comment (animeshsingh) [inline etc/examples/caffe-gpu-model/lenet_train_test.prototxt]: Same as above vis a vis duplication

File: etc/examples/caffe-gpu-model/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: mnist-caffe-gpu-model
```
Comment (animeshsingh) [inline etc/examples/caffe-gpu-model/manifest.yml]: Same as above - can`t these be instructions like change these params in manifest + proto files, and then run? This way users also learn

File: etc/examples/tf-gpu-model/input_data.py [hunk 4]
```diff
@@ -1,4 +1,3 @@
-
 #!/usr/bin/env python
```
Comment (animeshsingh) [inline etc/examples/tf-gpu-model/input_data.py]: Same duplication issue...?

**Comment by Tomcli** [comment]
@animeshsingh For the first three comments I think you are looking at the older commit. Regrading to the duplication, I can merge the cpu and gpu example together and give a more detailed instructions in the gpu-guide.md.

**Comment by Tomcli** [comment]
I put the CPU and GPU in a single example, so for TensorFlow and Caffe, there will be an extra gpu-manifest.yml to guide the user how to deploy with GPU resources. Then, the detailed instructions on converting CPU jobs to GPU is available at [gpu-guide](https://github.com/IBM/FfDL/blob/new-frameworks/docs/gpu-guide.md).

**Pull request merged**